python/common/org/Python.java

```diff
@@ -1005,10 +1005,24 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
         __doc__ = "next(iterator[, default])" +
             "\n" +
             "Return the next item from the iterator. If default is given and the iterator\n" +
-            "is exhausted, it is returned instead of raising StopIteration.\n"
+            "is exhausted, it is returned instead of raising StopIteration.\n",
+        args = {"iterator"},
+        default_args = {"_default"}
     )
-    public static org.python.Object next() {
-        throw new org.python.exceptions.NotImplementedError("Builtin function 'next' not implemented");
+    public static org.python.Object next(org.python.Object iterator, org.python.Object _default) {
+        if (iterator instanceof org.python.Iterable) {
```

Comment:
`instanceof` works; but if we can wrap this into the `utils.isinstance` mechanism, that would be even better.